### ISSUE TITLE
Type_int_key don't convert string to integer for elasticsearch

### ISSUE THREAD

**Issue by cde0** [fluent/fluent-bit#1932]
## Bug Report

**Describe the bug**
I have a lua script for parsing and then send to elastic. I tried to convert string field to integer by lua and send to elastic, but elastic show text field.

**To Reproduce**

- fluentd-bit.conf
`Type_int_key  id record["id"] uuid`

- elasticsearch return
`id" : {
     "type" : "text"
`
Also for uuid

**Expected behavior**
Expected behavior converts text to an integer.

**Your Environment**
* Version used:  td-agent-bit: 1.3.6
* Operating System and version: Ubuntu 18.04.3 LTS

**Comment by PettitWesley** [comment]
@cde0 Can you provide your full `fluent-bit.conf` config file and your lua script? An example log message would help too.

**Comment by cde0** [comment]
I add template mapping to elastic and is working ok now.

- Elastic return

` id" : { "type" : "long" `

- lua script is this: 

```
record["id"] = record["log"]:match("id=(%d+)",11)
 return 1, timestamp, record
```

- fluent-bit.conf 
is nothing special 

```
[FILTER]
    Name lua
    Match match_id.log
    script /opt/match_id.lua
    call match_id
    Type_int_key id record["id"] uuid
```

**Comment by nokute78** [comment]
@cde0 Hi.
It may be proper to use Lua function `tonumber` when you convert `string` to `integer`
http://www.lua.org/manual/5.1/manual.html#pdf-tonumber

`Type_int_key` is to prevent converting `integer` to `double`.

## example

This is an example to convert string `"100"` to integer `100`.

a.conf:
```
[INPUT]
    Name dummy
    dummy {"idstr":"100"}

[FILTER]
    Name lua
    Match *
    script conv.lua
    call conv
    Type_int_key id

[OUTPUT]
    Name stdout
```

conv.lua
```lua
function conv(tag, timestamp, record)
    record["id"] = tonumber(record["idstr"])
    return 1, timestamp, record
end
```

output:
```
taka@ubuntu:~/git/fluent-bit/build/1932$ ../bin/fluent-bit -c a.conf 
Fluent Bit v1.5.0
* Copyright (C) 2019-2020 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2020/04/08 09:35:51] [ info] [storage] version=1.0.3, initializing...
[2020/04/08 09:35:51] [ info] [storage] in-memory
[2020/04/08 09:35:51] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2020/04/08 09:35:51] [ info] [engine] started (pid=15636)
[2020/04/08 09:35:51] [ info] [sp] stream processor started
[0] dummy.0: [1586306152.001024484, {"id"=>100, "idstr"=>"100"}]
[1] dummy.0: [1586306153.000517845, {"id"=>100, "idstr"=>"100"}]
[2] dummy.0: [1586306154.000271558, {"id"=>100, "idstr"=>"100"}]
[3] dummy.0: [1586306155.000405550, {"id"=>100, "idstr"=>"100"}]
^C[engine] caught signal (SIGINT)
taka@ubuntu:~/git/fluent-bit/build/1932$ 
```

**Comment by edsiper** [comment]
the root cause of the problem has been fixed in this commit: c8f55c8b7850ca07c51002cfa2040642111eee18

the solution will be part of the next release Fluent Bit v1.5.7
